### PR TITLE
pyupgrade to 3.7

### PR DESCRIPTION
**Pull request by mr-c**
plus import sorting

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Misc: code updated to be more idiomatic for Python3.7

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/jobStores/fileJobStore.py [hunk 1]
```diff
@@ -650,7 +650,6 @@ def read_file_stream(
         else:
             with open(
                 self._get_file_path_from_id(file_id),
-                "rt",
                 1,
```
Comment (mr-c) [inline src/toil/jobStores/fileJobStore.py:653]:
```suggestion
                buffering=1,  # line buffering
```

File: src/toil/test/docs/scripts/example_cachingbenchmark.py [hunk 2]
```diff
@@ -51,8 +51,8 @@ def main():
 def root(job, options):
     # Make a file
     with job.fileStore.writeGlobalFileStream() as (stream, file_id):
-        stream.write(("This is a test of the Toil file caching system. "
-                      "Had this been an actual file, its contents would have been more interesting.").encode('utf-8'))
+        stream.write(b"This is a test of the Toil file caching system. "
+                      b"Had this been an actual file, its contents would have been more interesting.")
```
Comment (adamnovak) [inline src/toil/test/docs/scripts/example_cachingbenchmark.py:55]:
```suggestion
                     b"Had this been an actual file, its contents would have been more interesting.")
```

File: src/toil/test/src/importExportFileTest.py [hunk 3]
```diff
@@ -144,5 +144,5 @@ def run(self, file_store):
             with file_store.readGlobalFileStream(self.msg_portion_file_id, encoding='utf-8') as readable:
                 # combine readable.read() (the original message 1) with message 2
                 # this will be the final output of the workflow
-                writable.write(f'{readable.read()}{self.message_portion_2}'.encode('utf-8'))
+                writable.write(f'{readable.read()}{self.message_portion_2}'.encode())
```
Comment (adamnovak) [inline src/toil/test/src/importExportFileTest.py:147]: How come it wants to remove explicit `'utf-8'` encodings when we encode stuff?

File: version_template.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def exactPython():
     for. Something like 'python2.7' or 'python3.6'.
     """
     import sys
-    return 'python{}.{}'.format(sys.version_info[0], sys.version_info[1])
+    return f'python{sys.version_info[0]}.{sys.version_info[1]}'
```
Comment (adamnovak) [inline version_template.py:58]: Is this really clearer this way?

**Review by adamnovak — APPROVED** [review]
This looks good except where it has moved some code away from its comments, which I think I can fix.

File: src/toil/test/src/importExportFileTest.py (hunk 3 above)
Comment (mr-c) [reply src/toil/test/src/importExportFileTest.py:147]: That's the default encoding from a certain version of Python onward: https://github.com/asottile/pyupgrade/pull/485

File: version_template.py (hunk 4 above)
Comment (mr-c) [reply version_template.py:58]: It is more compact; most IDEs will make this fairly easy to read

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thank you @mr-c .

**Pull request merged**